File: RefugeUA.Entities/AnnouncementResponse.cs
using RefugeUA.Entities.Abstracts;
using RefugeUA.Entities.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefugeUA.Entities
{
    public class AnnouncementResponse : Entity
    {
        public AnnouncementResponse()
        {
            this.CreatedAt = DateTime.Now;
        }

        public long? UserId { get; set; }

        public IUser? User { get; set; } = default!;

        public long AnnouncementId { get; set; }

        public Announcement Announcement { get; set; } = default!;

        public long ContactInformationId { get; set; }

        public ContactInformation ContactInformation { get; set; } = default!;

        public DateTime CreatedAt { get; set; }
    }
}

File: RefugeUA.WebApp/refugeua.webapp.client/src/app/core/models/announcement-response.ts
import { Entity } from "./entity"
import { Announcement } from "./announcement"
import { ContactInformation } from "./contact-information"
import { User } from "./user"

export interface AnnouncementResponse extends Entity{
    userId?: number,
    user?: User,
    announcementId: number,
    announcement?: Announcement,
    contactInformationId: number,
    contactInformation?: ContactInformation,
    createdAt: Date 
}